phac-nml/irida-next
# Namespace group link UI

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI component of the Namespace to Group link. The members page for both groups and projects now displays 2 tabs (Members and Groups) with the Members tab listing all the individual members that are added directly or inherited, and the Groups tab lists all the groups that the namespace has been shared with. Also, the UI for sharing a namespace with a group has been setup with the modal being launched via the members pages.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Group Members**

![image](https://github.com/phac-nml/irida-next/assets/25466211/6ebf6016-f3b3-40a9-ac6b-4beb182bab29)

![image](https://github.com/phac-nml/irida-next/assets/25466211/0109ef2b-ac05-419c-864a-d1c9e6af365f)

**Group to Group Invite**

![image](https://github.com/phac-nml/irida-next/assets/25466211/1c9a1b9f-9ea7-4036-8d4b-89e7b2638aba)

**Project Members**

![image](https://github.com/phac-nml/irida-next/assets/25466211/11b84ecf-2551-47e5-acf8-7faa99a5426d)

![image](https://github.com/phac-nml/irida-next/assets/25466211/ffc5fd28-1b8e-4740-a414-1f3cc8aa6ad5)

**Project to Group Invite**

![image](https://github.com/phac-nml/irida-next/assets/25466211/aef177ef-f81e-423e-b52c-2390b05c21db)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as a user that has groups and projects
2. Select any group and view it's `Members` page
3. You should see the `Members` page with 2 tabs (Members, Groups) and a new button for `Invite Group`
4. Click the `Invite Group` button and a modal should appear for you to invite a group
5. Select a group to share the namespace with, select a group access level, and optional expiration
6. Click the `Share` button and the group should now be listed and the pagination info should be updated
7. From the listing page you can update a namespace group link's access level and expiration
8. Click the `Unlink` button for namespace group link. If a namespace group link was inherited and was added as a direct member, then unlinking the direct member will display the inherited share in the table
9. Please verify that all this functionality is working as expected, and in addition please ensure the members tab also is still working as expected
9. 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

  the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.
